### PR TITLE
hyprctl: change monitorRequest to report all connected monitors

### PR DESCRIPTION
**Pull request by TheAngusMcFire**
#### Describe your PR, what does it fix/add?
I added a way to show all monitors, not just the active ones.
I use this to see which monitor is connected to which output before changing the config. 

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
nope

#### Is it ready for merging, or does it need work?
ready for merge

### CODE REVIEW

**Comment by TheAngusMcFire** [comment]
@vaxerski is there anything else I have to to in order to get the PR merged?

**Comment by Eclextic** [comment]
> @vaxerski is there anything else I have to to in order to get the PR merged?

Request the review then...

File: src/debug/HyprCtl.cpp [hunk 1]
```diff
@@ -47,7 +60,13 @@ std::string monitorsRequest(HyprCtl::eHyprCtlOutputFormat format) {
     "serial": "{}",
     "width": {},
     "height": {},
-    "refreshRate": {:.5f},
+    "refreshRate": {:.5f})#",
+                m->ID, escapeJSONStrings(m->szName), escapeJSONStrings(m->output->description ? m->output->description : ""), (m->output->make ? m->output->make : ""),
+                (m->output->model ? m->output->model : ""), (m->output->serial ? m->output->serial : ""), (int)m->vecPixelSize.x, (int)m->vecPixelSize.y, m->refreshRate);
+
+            if (!allMonitors) {
```
[the remark this answers is not in the document]
Comment (vaxerski) [reply src/debug/HyprCtl.cpp:67]: it's better to print what's available. JSON-reading apps can easily handle missing props

File: src/debug/HyprCtl.cpp [hunk 2]
```diff
@@ -28,13 +28,24 @@ static std::string getWorkspaceNameFromSpecialID(const int workspaceID) {
     return workspace->m_szName;
 }
 
-std::string monitorsRequest(HyprCtl::eHyprCtlOutputFormat format) {
+std::string monitorsRequest(std::string request, HyprCtl::eHyprCtlOutputFormat format) {
+    CVarList vars(request, 0, ' ');
+    auto allMonitors = false;
+    
+    if (vars.size() > 2) {
+        return "to much args";
```
Comment (vaxerski) [inline src/debug/HyprCtl.cpp:36]: too many args

File: src/debug/HyprCtl.cpp [hunk 3]
```diff
@@ -28,13 +28,24 @@ static std::string getWorkspaceNameFromSpecialID(const int workspaceID) {
     return workspace->m_szName;
 }
 
-std::string monitorsRequest(HyprCtl::eHyprCtlOutputFormat format) {
+std::string monitorsRequest(std::string request, HyprCtl::eHyprCtlOutputFormat format) {
+    CVarList vars(request, 0, ' ');
+    auto allMonitors = false;
+    
+    if (vars.size() > 2) {
```
Comment (vaxerski) [inline src/debug/HyprCtl.cpp:35]: no {} around short ifs

File: src/debug/HyprCtl.cpp [hunk 4]
```diff
@@ -28,13 +28,24 @@ static std::string getWorkspaceNameFromSpecialID(const int workspaceID) {
...
+        return "to much args";
+    }
+
+    if (vars.size() == 2 && vars[1] == "all") {
```
Comment (vaxerski) [inline src/debug/HyprCtl.cpp:39]: ditto

**Review by vaxerski — COMMENTED** [review]
apart from that lgtm, also needs a wiki mr btw

**Comment by vaxerski** [comment]
too not to...

**Comment by TheAngusMcFire** [comment]
I created a wiki MR for the changes I made
https://github.com/hyprwm/hyprland-wiki/pull/396

File: src/debug/HyprCtl.cpp [hunk 5]
```diff
@@ -28,13 +28,22 @@ static std::string getWorkspaceNameFromSpecialID(const int workspaceID) {
     return workspace->m_szName;
 }
 
-std::string monitorsRequest(HyprCtl::eHyprCtlOutputFormat format) {
+std::string monitorsRequest(std::string request, HyprCtl::eHyprCtlOutputFormat format) {
+    CVarList vars(request, 0, ' ');
+    auto allMonitors = false;
+    
+    if (vars.size() > 2)
+        return "to many args";
```
Comment (vaxerski) [inline src/debug/HyprCtl.cpp:36]: you still havent fixed this typo. "too many args" not "to many args"

Comment (TheAngusMcFire) [reply src/debug/HyprCtl.cpp]: Oh, sorry I overlooked that one :see_no_evil: , I fixed it now.

**Review by vaxerski — APPROVED** [review]
lgtm

**Pull request merged**